logproof/src/linear_relation.rs

```diff
@@ -1404,4 +1409,329 @@ mod test {
     fn transcripts_match_k_4() {
         transcripts_match(4);
     }
+
+    /**
+     * Remove an element trailing in a vector. This can be helpful for types
+     * like `DensePolynomial`, which do not work properly if the polynomials
+     * passed in have a leading polynomial coefficient of zero.
+     */
+    fn strip_trailing_value<T>(mut v: Vec<T>, trim_value: T) -> Vec<T>
+    where
+        T: Eq,
+    {
+        while v.last().map_or(false, |c| *c == trim_value) {
+            v.pop();
+        }
+
+        v
+    }
+
+    /**
+     * Converts a polynomial known to have coefficients less than all of the
+     * moduli in its associated modulus set into regular integers. The main
+     * advantage here over using a polynomial in its normal field is that the
+     * polynomial can be moved to a new field without modulus switching.
+     */
+    #[allow(unused)]
+    fn convert_to_smallint(
```

Comment:
These small int functions are not used anymore but I have left them in case we need them later. The original idea was that I wanted to test that the small integer representation, which can be easier to work with in some cases, was equivalent to the full RNS representation. These may be useful in later computations where we are performing calculations in different fields from the field SEAL is using.